configure.ac

```diff
@@ -738,9 +738,9 @@ else
 		;;
 	esac
 fi
-CFLAGS="$CFLAGS -g $WARN"
+CFLAGS="$WARN $CFLAGS -g"
 CFLAGS_FOR_LIBGC="$CFLAGS_FOR_LIBGC -g"
```

Comment:
Why is this needed ? The usual way of adding flags is to append to them.

Reply:
Otherwise we have the following command line being executed for `pal_io.c` for example:

```
echo "  CC      " ../../external/corefx/src/Native/Unix/System.Native/libmonoruntime_sysnat_la-pal_io.lo;../../doltlibtool --silent --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I/Users/luhenry/git/mono/external/corefx/src/Native/Unix/Common -I/Users/luhenry/git/mono/external/corefx/src/Native/Unix/System.Native -Wno-typedef-redefinition  -Wno-missing-prototypes -D_THREAD_SAFE -DGC_MACOSX_THREADS -DUSE_MMAP -DUSE_MUNMAP -g -Wall -Wunused -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes  -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wno-cast-qual -Wwrite-strings -Wno-switch -Wno-switch-enum -Wno-unused-value -Wno-attributes -Wno-format-zero-length -Qunused-arguments -Wno-unused-function -Wno-tautological-compare -Wno-parentheses-equality -Wno-self-assign -Wno-return-stack-address -Wno-constant-logical-operand -Wno-zero-length-array   -O0 -g -std=gnu99 -fno-strict-aliasing -fwrapv -DMONO_DLL_EXPORT -g -Wall -Wunused -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes  -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wno-cast-qual -Wwrite-strings -Wno-switch -Wno-switch-enum -Wno-unused-value -Wno-attributes -Wno-format-zero-length -Qunused-arguments -Wno-unused-function -Wno-tautological-compare -Wno-parentheses-equality -Wno-self-assign -Wno-return-stack-address -Wno-constant-logical-operand -Wno-zero-length-array -Werror-implicit-function-declaration -MT ../../external/corefx/src/Native/Unix/System.Native/libmonoruntime_sysnat_la-pal_io.lo -MD -MP -MF ../../external/corefx/src/Native/Unix/System.Native/.deps/libmonoruntime_sysnat_la-pal_io.Tpo -c -o ../../external/corefx/src/Native/Unix/System.Native/libmonoruntime_sysnat_la-pal_io.lo `test -f '../../external/corefx/src/Native/Unix/System.Native/pal_io.c' || echo './'`../../external/corefx/src/Native/Unix/System.Native/pal_io.c
  CC       ../../external/corefx/src/Native/Unix/System.Native/libmonoruntime_sysnat_la-pal_io.lo
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:180:19: warning: 'stat64' is deprecated: first deprecated in macOS 10.6 [-Wdeprecated-declarations]
    while ((ret = stat_(path, &result)) < 0 && errno == EINTR);
                  ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:41:15: note: expanded from macro 'stat_'
#define stat_ stat64
              ^
/usr/include/sys/stat.h:402:5: note: 'stat64' has been explicitly marked deprecated here
int     stat64(const char *, struct stat64 *) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_5,__MAC_10_6,__IPHONE_NA,__IPHONE_NA);
        ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:176:9: warning: no previous prototype for function 'SystemNative_Stat2' [-Wmissing-prototypes]
int32_t SystemNative_Stat2(const char* path, struct FileStatus* output)
        ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:194:19: warning: 'fstat64' is deprecated: first deprecated in macOS 10.6 [-Wdeprecated-declarations]
    while ((ret = fstat_(ToFileDescriptor(fd), &result)) < 0 && errno == EINTR);
                  ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:42:16: note: expanded from macro 'fstat_'
#define fstat_ fstat64
               ^
/usr/include/sys/stat.h:400:5: note: 'fstat64' has been explicitly marked deprecated here
int     fstat64(int, struct stat64 *) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_5,__MAC_10_6,__IPHONE_NA,__IPHONE_NA);
        ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:190:9: warning: no previous prototype for function 'SystemNative_FStat2' [-Wmissing-prototypes]
int32_t SystemNative_FStat2(intptr_t fd, struct FileStatus* output)
        ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:207:15: warning: 'lstat64' is deprecated: first deprecated in macOS 10.6 [-Wdeprecated-declarations]
    int ret = lstat_(path, &result);
              ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:43:16: note: expanded from macro 'lstat_'
#define lstat_ lstat64
               ^
/usr/include/sys/stat.h:401:5: note: 'lstat64' has been explicitly marked deprecated here
int     lstat64(const char *, struct stat64 *) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_5,__MAC_10_6,__IPHONE_NA,__IPHONE_NA);
        ^
../../external/corefx/src/Native/Unix/System.Native/pal_io.c:204:9: warning: no previous prototype for function 'SystemNative_LStat2' [-Wmissing-prototypes]
int32_t SystemNative_LStat2(const char* path, struct FileStatus* output)
        ^
6 warnings generated.
```

What you can note is we define `-Wno-missing-prototypes` in `libmonoruntime_sysnat_la_CPPFLAGS` but it gets overwritten by `$WARN` at https://github.com/mono/mono/pull/8051#discussion-diff-179934409L741. By changing the order of `$CFLAGS` and `$WARN`, we don't get this warning anymore.